### ISSUE TITLE
Error encountered rendering book

### ISSUE THREAD

**Issue by hermandr** [moderndive/moderndive_book#146]
Hi,

I am trying to render the book in html_book using the following command

bookdown::render_book("index.Rmd", "bookdown::html_book")

I encountered the following error:
Error in split_chapters(output, page_builder, number_sections, split_by,  : 
  The document must start with a first (#) or second level (##) heading
In addition: There were 50 or more warnings (use warnings() to see the first 50)

What am I doing wrong?

Herman

**Comment by rudeboybert** [comment]
Hi @hermandr, sorry for the delay in replying. Could you run `sessionInfo()` in R and copy the output in code format here?

**Comment by hermandr** [comment]
output file: ismaykim.knit.md

"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS ismaykim.utf8.md --to html4 --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash+smart --output ismaykim.html --email-obfuscation none --wrap preserve --standalone --section-divs --table-of-contents --toc-depth 3 --template "C:\Users\herma\Documents\R\win-library\3.5\bookdown\templates\default.html" --highlight-style pygments --number-sections --variable "theme:bootstrap" --include-in-header "C:\Users\herma\AppData\Local\Temp\RtmpEjsKAJ\rmarkdown-stre745df6c7f.html" --mathjax --filter pandoc-citeproc 
Error in split_chapters(output, page_builder, number_sections, split_by,  : 
  The document must start with a first (#) or second level (##) heading
In addition: There were 50 or more warnings (use warnings() to see the first 50)
> sessionInfo()
R version 3.5.3 (2019-03-11)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

locale:
[1] LC_COLLATE=English_Singapore.1252  LC_CTYPE=English_Singapore.1252    LC_MONETARY=English_Singapore.1252
[4] LC_NUMERIC=C                       LC_TIME=English_Singapore.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] skimr_1.0.5           dygraphs_1.1.1.6      scales_1.0.0          ggplot2movies_0.0.1  
 [5] infer_0.4.0           ISLR_1.2              patchwork_0.0.1       janitor_1.2.0        
 [9] broom_0.5.1           gridExtra_2.3         forcats_0.4.0         mvtnorm_1.0-10       
[13] moderndive_0.2.0      stringr_1.4.0         fivethirtyeight_0.4.0 tidyr_0.8.3          
[17] readr_1.3.1           kableExtra_1.0.1      gapminder_0.3.0       ggplot2_3.1.1.9000   
[21] knitr_1.22            dplyr_0.8.0.1         nycflights13_1.0.0   

loaded via a namespace (and not attached):
 [1] nlme_3.1-137               xts_0.11-2                 assertive.models_0.0-2    
 [4] webshot_0.5.1              httr_1.4.0                 assertive.datetimes_0.0-2 
 [7] tools_3.5.3                backports_1.1.3            utf8_1.1.4                
[10] R6_2.4.0                   lazyeval_0.2.1             colorspace_1.4-0          
[13] assertive.data_0.0-3       withr_2.1.2                assertive.reflection_0.0-4
[16] tidyselect_0.2.5           curl_3.3                   compiler_3.5.3            
[19] cli_1.0.1                  rvest_0.3.2                assertive.properties_0.0-4
[22] xml2_1.2.0                 labeling_0.3               assertive.files_0.0-2     
[25] bookdown_0.9               digest_0.6.18              rmarkdown_1.11.11         
[28] assertive.numbers_0.0-2    pkgconfig_2.0.2            htmltools_0.3.6           
[31] highr_0.7                  htmlwidgets_1.3            rlang_0.3.1               
[34] rstudioapi_0.9.0           assertive_0.3-5            generics_0.0.2            
[37] zoo_1.8-4                  jsonlite_1.6               magrittr_1.5              
[40] Rcpp_1.0.0                 munsell_0.5.0              fansi_0.4.0               
[43] stringi_1.4.3              assertive.base_0.0-7       yaml_2.2.0                
[46] snakecase_0.9.2            plyr_1.8.4                 grid_3.5.3                
[49] formula.tools_1.7.1        crayon_1.3.4               lattice_0.20-38           
[52] assertive.code_0.0-3       hms_0.4.2.9001             pillar_1.3.1              
[55] assertive.sets_0.0-3       reshape2_1.4.3             codetools_0.2-16          
[58] glue_1.3.0                 packrat_0.5.0              evaluate_0.13             
[61] operator.tools_1.6.3       gtable_0.2.0               purrr_0.3.1               
[64] assertive.strings_0.0-3    assertthat_0.2.0           xfun_0.5                  
[67] assertive.types_0.0-3      assertive.data.uk_0.0-2    rsconnect_0.8.13          
[70] viridisLite_0.3.0          tibble_2.1.1               assertive.matrices_0.0-2  
[73] assertive.data.us_0.0-2

**Comment by hermandr** [comment]
When I knit from RStudio I only see the html generated even when I choose knit to pdf.

**Comment by Hendrik147** [comment]
Herman, 
when I click on the knit button with index.rmd open, it creates automatically the pdf, but it does not create the various html files in the docs folder.
Can you compare please your index.rmd file with mine. It is located here: https://github.com/Hendrik147/HR_Analytics_in_R_book
Hendrik

**Comment by Hendrik147** [comment]
New things I discovered since my previous post: 

When `bookdown::pdf_book:` appears first in `_output.yml` and when one clicks on the knit button in Rstudio, it only knits the first instance in `_output.yml`.

Alternatively one can just type the commands below in the R console:
`bookdown::render_book('index.Rmd', 'bookdown::gitbook')` or
`bookdown::render_book('index.Rmd', 'bookdown::pdf_book')`

The bookdown book (https://bookdown.org/yihui/bookdown) has documented these commands.

**Comment by rudeboybert** [comment]
Hi @hermandr, I was able to generate the PDF by following @Hendrik147's suggestions. Were you?

**Comment by hermandr** [comment]
Hendrik and Albert,

Thank you for the tips.

It generated the md file but fails to created the pdf file failing with the following errors.

`output file: ismaykim.knit.md

"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS ismaykim.utf8.md --to latex --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output ismaykim.tex --table-of-contents --toc-depth 2 --template "C:\Users\herma\Documents\R\win-library\3.5\rmarkdown\rmd\latex\default-1.17.0.2.tex" --number-sections --highlight-style tango --pdf-engine pdflatex --variable graphics=yes --wrap preserve --variable "geometry:margin=1in" --variable "compact-title:yes" --include-in-header "C:\Users\herma\AppData\Local\Temp\RtmpEnmOOT\rmarkdown-str4dd47c633a1d.html" --variable tables=yes --standalone --filter pandoc-citeproc 
tlmgr search --file --global "/multirow.sty"
Trying to automatically install missing LaTeX packages...
tlmgr install multirow
TeX Live 2018 is frozen forever and will no
longer be updated.  This happens in preparation for a new release.

If you're interested in helping to pretest the new release (when
pretests are available), please read http://tug.org/texlive/pretest.html.
Otherwise, just wait, and the new release will be ready in due time.
===============================================================================
tlmgr itself needs to be updated.
Please do this via either
  tlmgr update --self
or by getting the latest updater for Unix-ish systems:
  http://mirror.ctan.org/systems/texlive/tlnet/update-tlmgr-latest.sh
and/or Windows systems:
  http://mirror.ctan.org/systems/texlive/tlnet/update-tlmgr-latest.exe
Then continue with other updates as usual.
===============================================================================
tlmgr.pl: Terminating; please see warning above!
tlmgr.pl: package repository http://mirror.kku.ac.th/CTAN/systems/texlive/tlnet (not verified: gpg unavailable)
tlmgr search --file --global "/multirow.sty"
! LaTeX Error: File `multirow.sty' not found.

! Emergency stop.
<read *> 

Error: Failed to compile ismaykim.tex. See https://yihui.name/tinytex/r/#debugging for debugging tips. See ismaykim.log for more info.
In addition: There were 50 or more warnings (use warnings() to see the first 50)`

I have also attached the `ismaykim.log`.

[ismaykim.log](https://github.com/moderndive/moderndive_book/files/3230660/ismaykim.log)

**Comment by hermandr** [comment]
I have installed
> http://mirror.ctan.org/systems/texlive/tlnet/update-tlmgr-latest.exe

After running it with 
`bookdown::render_book('index.Rmd', 'bookdown::pdf_book')`

I get the following error:
output file: ismaykim.knit.md

"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS ismaykim.utf8.md --to latex --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output ismaykim.tex --table-of-contents --toc-depth 2 --template "C:\Users\herma\Documents\R\win-library\3.5\rmarkdown\rmd\latex\default-1.17.0.2.tex" --number-sections --highlight-style tango --pdf-engine pdflatex --variable graphics=yes --wrap preserve --variable "geometry:margin=1in" --variable "compact-title:yes" --variable tables=yes --standalone --filter pandoc-citeproc 
! LaTeX Error: Environment announcement undefined.

Error: Failed to compile ismaykim.tex. See https://yihui.name/tinytex/r/#debugging for debugging tips. See ismaykim.log for more info.
In addition: There were 50 or more warnings (use warnings() to see the first 50)

I have attached the logfile as well.
[ismaykim.log](https://github.com/moderndive/moderndive_book/files/3230696/ismaykim.log)

**Comment by hermandr** [comment]
I found that "announcement" environment was undefined.
I deleted all references to "announcement" in the Rmd.

Now I find that "learncheck" environment is undefined.

I found the preamble.tex in latex folder containing 
`\newenvironment{learncheck}
  {\begin{rmdblock}{warning}}
  {\end{rmdblock}}`

It seems that during the compile to pdf phase, it is unable to resolve "learncheck" environment and probably did not load the preamble .tx file in the tex folder.
Something is wrong with the setup for pdf output.
